charts/kube-prometheus-stack/README.md

```diff
@@ -200,6 +200,56 @@ The prometheus operator does not support annotation-based discovery of services,
 
 By default, Prometheus discovers ServiceMonitors within its namespace, that are labeled with the same release tag as the prometheus-operator release. Sometimes, you may need to discover custom ServiceMonitors, for example used to scrape data from third-party applications. An easy way of doing this, without compromising the default ServiceMonitors discovery, is allowing Prometheus to discover all ServiceMonitors within its namespace, without applying label filtering. To do so, you can set `prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues` to `false`.
 
+## Migrating from stable/prometheus-operator chart
+
+If the **prometheus-operator** values are compatible with the new **kube-prometheus-stack** chart, please follow the below steps for migration:
+
+> The guide presumes that chart is deployed in `monitoring` namespace and the deployments are running there. If in other namespace, please replace the `monitoring` to the deployed namespace.
+
+1. Patch the PersistenceVolume created/used by the prometheus-operator chart to `Retain` claim policy:
+
+```bash
+kubectl patch pv/<PersistentVolume name> -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}' -n monitoring
+```
+
+**Note:** To execute the above command, the user must have a cluster wide permission. Please refer [Kubernetes RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
+
+2. Uninstall the **prometheus-operator** release and delete the existing PersistentVolumeClaim, and verify PV become Released.
+
+```bash
+helm uninstall prometheus-operator -n monitoring
+kubectl delete pvc/<PersistenceVolumeClaim name> -n monitoring
+```
+
+Additonaly, you have to manually remove the remaining `prometheus-operator-kubelet` service.
+
+```bash
+kubectl delete service/prometheus-operator-kubelet -n kube-system
+```
+
+3. Remove current `spec.claimRef` values to change the PV's status from Released to Available.
```

Comment:
Could some other unrelated PVC, created in between deletion of old PVC and creation of new one, take control of this PV?
Usually, when I do this kind of migration, I delete the old PVC, create the new one with correct PV reference, and edit the PV to point to the new PV.

Reply:
It sounds like this has been discussed below, but if there's a recommendation for mitigating this further let's update these instructions in a follow-up PR. Right now, there is no good documented migration path so this is an improvement.